base/deprecated.jl

```diff
@@ -1093,6 +1093,24 @@ end
 #15995
 @deprecate symbol Symbol
 
+#15032: Expressions like Base.(:+) now call broadcast.  Since there is unlikely
+#       to be any legitimate reason to broadcast to a Module (not iterable!),
+#       it should be safe to catch this with a deprecation method.
+#       Note: this deprecation does not handle method definitions Base.(:+)(...) = ...,
+#             which are handled by the deprecate-dotparen function in julia-syntax.scm
+#       More generally, expressions like x.(sym) for any Symbol turn into broadcast
+#       but should really be getfield, and since you can't broadcast to a symbol
+#       using a deprecated broadcast method should be safe.   [Unfortunately,
+#       we can't catch x.(i::Int) in the same way, because broadcasting to an
```

Comment:
Yes. I would further say that while there are several uses of `a.(i::Int)` as getfield, I highly doubt there is even a single use of `broadcast` over an integer. So I think it would be better to give a deprecation warning here.